### ISSUE TITLE
Interfaces should not be under `#if`

### ISSUE THREAD

**Issue by Qbit86** [playgameservices/play-games-plugin-for-unity#1179]
After upgrading from 0.9.27 to 0.9.33 project fails to build for Windows Phone (`UNITY_WSA`). This is so because some interfaces used in my method params disappeared under `#if (UNITY_ANDROID || (UNITY_IPHONE && !NO_GPGS))`. For example, `ISavedGameMetadata`.

In general, public interfaces and methods signatures should not be under conditional compilation — instead their implementations should be.

It's better when `#if`-wrapping does not occur in caller side, but rather inside the the callee side. It's ok to wrap platform-specific implementation details, but interfaces should be as much cross-platform as possible.

**Comment by claywilkinson** [comment]
Thanks for the feedback.  How did you fix it in your case?

**Comment by Qbit86** [comment]
> How did you fix it in your case?

I had to revert these new compilation conditions for some interfaces/enums in source code of plugin:
```
// #if (UNITY_ANDROID || (UNITY_IPHONE && !NO_GPGS))
...
    public interface ISavedGameMetadata
```
A lot of interfaces are “crossplatform-safe” (depending only on standard types), and their usage is “platform-agnostic”.

In my code I try to avoid conditional compilation, in preference to runtime checks / polymorphism. Excluding code from compilation may be little more efficient than runtime checks, but is less maintainable. It complicates refactoring (renaming and so on), which is better to be AST-based in IDE, not lexical-based in plain text editor. And preprocessor hides code from compiler, prevents building rich context-aware syntax tree.

**Comment by claywilkinson** [comment]
ISavedGameMetadata is fixed, as well as some other interfaces in 0.9.34.  The multiplayer listener interfaces are not yet because they require Player, and Player has implementation that requires other code.  Once Player is refactored, the other interfaces can be fixed also.
